documentation/release_notes.markdown

```diff
@@ -14,6 +14,108 @@ canonical: "/puppetdb/latest/release_notes.html"
 [metrics]: ./api/metrics/v1/changes-from-puppetdb-v3.html
 [pqltutorial]: ./api/query/tutorial-pql.html
 
+4.2.0
+-----
+
+PuppetDB 4.2.0 is a backward-compatible feature release that adds a
+new inventory endpoint, the ability to query structured data like
+facts and resource parameters using a dotted notation and the ability
+to trigger background/GC tasks manually via POST. This release
+also includes several new queryable fields, bugfixes and faster
+retries for common error scenarios.
+
+### New features / Enhancements
+
+* Added a new inventory endpoint that includes all fact keypairs
+  represented as a map. Trusted facts are also returned as a map in a
+  separate keypair. This new data can be queried via our "dotted
+  syntax" in both AST and PQL query languages. More information is
+  available via the [API docs](./api/query/v4/inventory.html).
+  ([PDB-2632](https://tickets.puppetlabs.com/browse/PDB-2632))
+
+* Added support for "dotted query" syntax to resource
+  parameters. Support for this is available in both AST and PQL query
+  langauges. ([PDB-2632](https://tickets.puppetlabs.com/browse/PDB-2632))
+
+* Added a new query parameter `ast_only` to the root query endpoint
+  that translates a PQL query to an equivalent AST query. More
+  information is available in the [root
+  endpoint](./api/query/v4/index.html#url-parameters) docs.
+  ([PDB-2528](https://tickets.puppetlabs.com/browse/PDB-2528))
+
+* Added "quick retries" on command failures. This will retry commands
+  (such as storing a new report or updating a catalog) immediately
+  after a failure. The command will be retried 4 times before falling
+  back to the regular retry process that involves delaying the message
+  and reenqueuing it. This should result in faster command processing
+  times and use less disk I/O in the most common kinds of failures.
+  ([PDB-2865](https://tickets.puppetlabs.com/browse/PDB-2865))
+
+* Added a new endpoint to manually trigger PuppetDB GC processes. The
+  new endpoint allows purging old reports or deactivated nodes via a
+  POST. Users can disable automatic running of these background
+  processes and schedule these manual triggers via cron to ensure they
+  don't disrupt peak usage times. More information on this new
```

Comment:
Wondered about something like:
"Added an endpoint that can trigger PuppetDB GC processes via a
POST (like purging old reports or deactivated nodes).  With that,
you can disable the automatic scheduling of those processes, and
run them on-demand, perhaps via cron, to avoid increasing
the load during peak usage times."